src/bridge/cockpitdbusmachines.c

```diff
@@ -0,0 +1,468 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+
+#include <glob.h>
+#include <string.h>
+#include <errno.h>
+#include <json-glib/json-glib.h>
+
+#include "cockpitdbusinternal.h"
+#include "common/cockpitconf.h"
+#include "common/cockpitsystem.h"
+
+#define MACHINES_SIG "a{sa{sv}}"
+
+/* counts the number of file change events that have not yet gotten a PropertiesChanged signal */
+static guint pending_updates;
+
+static const char *
+machines_json_dir (void)
```

Comment:
Ah, it would not only affect the globbing, but we'd also need multiple file monitors and we would still need to pick one that we use for writing into. This is a lot of production code complication just to support tests, so as discussed on IRC, I'll move this back to the original proposal of a single dir env var, but call it `$COCKPIT_TEST_CONFIG_DIR` in line with the existing `$COCKPIT_TEST_SERVER_PORT`.